### DISCUSSION TITLE
Exposing environmental variables only to main process

### DISCUSSION THREAD

**Discussion by martiinii** [cawa-93/vite-electron-builder#894] [Q&A] [answered]
Hey! How to expose environmental variables (without **VITE_** prefix) to main process in such a way, that they aren't visible in renderer processes?
I have database connection setup in main process but I want to have username/password stored in **.env** file.
From readme:
> Only VITE_SOME_KEY will be exposed as import.meta.env.VITE_SOME_KEY to your client source code, but DB_PASSWORD will not.

I thought that DB_PASSWORD will be exposed to non-client source code, like main process.
In main process I tried import.meta.env and process.env. I don't have really great knowledge of Vite, sorry for that.
How could this be implemented?

Thanks in advance!

**Comment by cawa-93** [comment, marked as answer]
You may customize prefixes. See [`envPrefix`](https://vitejs.dev/config/shared-options.html#envprefix). In your case, probably needed add two diferent prefixes: `['VITE_', 'MAIN_']`. `VITE_` for shared environmental variables, and `MAIN_` only for main process. Like `MAIN_DB_PASSWORD`
